### PR TITLE
WIP Feat/dynamic options

### PR DESCRIPTION
**Pull request by donaldwasserman**
Basically I did a little lightweight refactoring to make it a little easier to set properties. 

@FutoRicky Let me know your thoughts on this approach.

I plan on adding some tests maybe as well.

See #74

### CODE REVIEW

**Comment by donaldwasserman** [comment]
@FutoRicky Do all of these tests usually pass?

**Comment by FutoRicky** [comment]
@donaldwasserman Yessir! They should be passing, with this changes the tests need to be changed though I assume. Sorry I have been busy and haven't got the time to check out the PRs.

**Comment by donaldwasserman** [comment]
weird - the tests that actually test the functionality i changed seem to be passing 😄 

I'll git those fixed and let you know when those are done.

**Comment by FutoRicky** [comment]
Hey, I'm having some issues getting the addon to work with these changes. For some reason it's not detecting the url. any idea?

**Comment by donaldwasserman** [comment]
I honestly haven't had a chance to look at this more. I think i know what the problem is and will get all the tests passing and addon working this weekend/next week.

**Comment by FutoRicky** [comment]
Cool. I merged another PR so it would be wise to do a rebase.

**Comment by donaldwasserman** [comment]
@FutoRicky It took me a while to get back to this. All the tests should be green.

**Comment by donaldwasserman** [comment]
@FutoRicky Do you know why the phantom JS tests fail locally?

Otherwise, please let me know if this works for you.

**Comment by FutoRicky** [comment]
All tests are passing on my side, awesome! 

I'm unable to make the addon work correctly using dynamic options. Can you give me an example of how you are expecting the `config` attr to be set? 

I did something like this:

```
  options: Ember.computed(function() {
    return {
      url: '#',
      addRemoveLinks: true,
      autoProcessQueue: false,
      addedfile: this.addedfile
    };
  })
```

`{{drop-zone config=options}}`

When adding a file I get this error `Uncaught TypeError: b.apply is not a function`

**Comment by donaldwasserman** [comment]
so, obviously the words I chose were not "self documenting." 

I renamed some methods + added better comments. 

now config actually should be the right property.

File: addon/components/drop-zone.js [hunk 1]
```diff
@@ -104,9 +127,13 @@ export default Ember.Component.extend({
...
-
-  config: Ember.computed(function(){
-    let config = this.get('dzConfigHash'),
+  /**
+   * internal config cp
+   * @private
+   * @return {[type]} [description]
+   */
+  _dzConfig: Ember.computed(function(){
+    let config = this.get('confi'),
```
Comment (FutoRicky) [inline addon/components/drop-zone.js]: This may be the reason the test build is failing. It should be `let config = this.get('config'),` right?

**Comment by donaldwasserman** [comment]
Don't drink and vim

**Pull request merged**